[Migrator] Add fix-it tests to pick up swap exclusivity fix-its

bitjammer opened the pull request:
These are attached to warnings so they need to be added to the
opt-in list in the FixitFilter.

rdar://problem/31916085

bitjammer commented:
@swift-ci Please smoke test

bitjammer commented:
@swift-ci Please smoke test

bitjammer commented:
@swift-ci Please smoke test

devincoughlin commented on test/Migrator/swap_fixit.swift:
```diff
@@ -0,0 +1,30 @@
+// RUN: rm -rf %t && mkdir -p %t && %target-swift-frontend -emit-migrated-file-path %t/swap_fixit.swift.result -enforce-exclusivity=checked -swift-version 3 -emit-sil -primary-file %s -o /dev/null -verify
```
I don't think you want the '-verify' here (which is why the bots are complaining).

bitjammer replied:
Ah, that's what I get for copy/paste! Thanks!

devincoughlin reviewed (approved):
LGTM with the test change to remove '-verify'.

bitjammer commented:
@swift-ci Please smoke test

bitjammer committed:
[Migrator] Add fix-it tests to pick up swap exclusivity fix-its

These are attached to warnings so they need to be added to the
opt-in list in the FixitFilter.

rdar://problem/31916085

bitjammer commented:
Sorry for the testing noise, I'm currently trying to build other patches locally for something urgent.

bitjammer commented:
@swift-ci Please smoke test

bitjammer commented:
Hm, weird. It didn't pick up the fix-it. Building locally now.

bitjammer commented:
So, Devin tells me this is a SIL diagnostic, which is pretty obvious to me now considering I had to add the SILDiagnostics header. :-) It might be interesting to turn SIL generation on for the Migration pass but that significantly enlarges the scope of this patch.
